src/op_mode/nat-translation.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python3
+
+'''
+what we do and why
+'''
+
+import os
+import sys
+import ipaddress
+import argparse
+import xmltodict
+
+from vyos.util import popen
+from vyos.util import DEVNULL
+
+conntrack = '/usr/sbin/conntrack'
+
+verbose_format = "%-20s %-18s %-20s %-18s"
+normal_format = "%-20s %-20s %-4s  %-8s %s"
+
+
+def headers(verbose, pipe):
+    if verbose:
+        return verbose_format % ('Pre-NAT src', 'Pre-NAT dst', 'Post-NAT src', 'Post-NAT dst')
+    return normal_format % ('Pre-NAT', 'Post-NAT', 'Prot', 'Timeout', 'Type' if pipe else '')
+
+
+def command(srcdest, proto, ipaddr):
+    command = f'{conntrack} -o xml -L'
+
+    if proto:
+        command += f' -p {proto}'
+
+    if srcdest == 'source':
+        command += ' -n'
+        if ipaddr:
+            command += f' --orig-src {ipaddr}'
+    if srcdest == 'destination':
+        command += ' -g'
+
+    return command
+
+
+def run():
+    if not os.path.exists(conntrack):
```

Comment:
I don't think we should be testing this. If it's stated as a dependency in debian/control, then there's no chance it can be missing.

Reply:
I was just replicating "line for line" what the perl does - happy to drop :-)